services/alertmanager/alertmanager.go

```diff
@@ -743,77 +741,128 @@ func (svc *Service) GetAlerts(ctx context.Context) ([]*ammodels.GettableAlert, e
 	return resp.Payload, nil
 }
 
-// FindAlertByID searches alert by ID in alertmanager.
-func (svc *Service) FindAlertByID(ctx context.Context, id string) (*ammodels.GettableAlert, error) {
+// FindAlertsByID searches alerts by IDs in alertmanager.
+func (svc *Service) FindAlertsByID(ctx context.Context, ids []string) ([]*ammodels.GettableAlert, error) {
 	alerts, err := svc.GetAlerts(ctx)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get alerts form alertmanager")
 	}
 
+	m := make(map[string]struct{}, len(ids))
```

Comment:
`len` is done.

Regarding `map[string]struct{}` it's an idiomatic way in go to implement `set`. When you need only to check existence (i.e. you need only keys), the empty struct fits better. With `bool` you can face the question "What does this bool mean?". It can be existence, like in our example, but it can be something else. With `stuct{}` you will not have such kind of questions because there are no values at all. Also map of empty struct a little bit more performant.